kernel/arch/context.rs

```diff
@@ -289,20 +283,16 @@ pub unsafe fn context_clone(regs: &Regs) -> Result<usize> {
 
                 parent.files.clone()
             } else {
-                let mut files: Vec<ContextFile> = Vec::new();
-                for file in (*parent.files.get()).iter() {
-                    match file.resource.dup() {
-                        Ok(resource) => {
-                            //debugln!("{}: {}: dup resource {} for {}", parent.pid, parent.name, file.fd, clone_pid);
-
-                            files.push(ContextFile {
-                                fd: file.fd,
-                                resource: resource,
-                            });
-                        },
-                        Err(_err) => () //debugln!("{}: {}: failed to dup resource {} for {}: {}", parent.pid, parent.name, file.fd, clone_pid, err)
-                    }
-                }
+                let files: Vec<ContextFile> = (*parent.files.get())
+                    .iter()
+                    .filter(|file| file.resource.dup().is_ok())
+                    .map(|file|
+                        ContextFile {
+                            fd: file.fd,
+                            resource: file.resource.dup().unwrap(),
```

Comment:
I `filter()`'d the iterator beforehand to contain only items that have `Ok(resource)` when doing the dup. So it's safe to unwrap without panicking, if that's what you are referring to?